File: Javascript/Arrays/Insercion_inicio.js
/*
Arrays en Javascript,Insercion en el primer elemento de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 03/09/2026
*/

//declaracion de arreglo
const input_arr = [11, 21, 31, 41, 51, 61];
const elem = 16;

console.log("Antes de la insercion del arreglo:");
for (let i = 0; i < input_arr.length; i++){
    console.log(input_arr[i]);
}

input_arr.unshift(elem)     //inserta un nuevo elemento en el indice 0 y empuja a los demas

console.log("Despues de la insercion del arreglo:");
for (let i = 0; i < input_arr.length; i++){
    console.log(input_arr[i]);
}

File: C#/Arrays/Insercion_inicio.cs
/*
Arrays en C#,Insercion en el primer elemento de un array.
@Autor: Daniel Alejandro Alvarado Velázquez
@grupo: #2-02
@Fecha: 06/09/2026
*/

using System;

namespace Insercion_inicio
{
    class Program
    {
        static void Main(String[] args)
        {
            int [] input_arr = {1, 11, 21, 31, 41, 51};
            int elem = 27;

            Console.WriteLine("Antes de la insercion, el array es: ");
            foreach (int elemento in input_arr){
            Console.WriteLine(elemento);
            }

            int[] new_array = new int[input_arr.Length + 1];        //se creo un nuevo arreglo
            new_array[0] = elem;

            input_arr.CopyTo(new_array, 1);

            Console.WriteLine("Despues de la insercion, el array es: ");
            foreach (int elemento in new_array){
            Console.WriteLine(elemento);
            }


        }
    }
}